text/010-wasm-pack-watch-and-serve.md

```diff
@@ -0,0 +1,230 @@
+Game of Life tutorial][no-bundler-in-docs] without requiring readers to install
+any additional tools other than the Rust toolchain and `wasm-pack` when setting
+up their development environment. The goal here being less moving parts to learn
+and tools to wrangle when first exploring Rust and WebAssembly.
+
+Additionally, we would like to make sure that the local development server uses
+the `application/wasm` MIME type when serving Wasm binaries. This enables the
+`WebAssembly.instantiateStreaming` fast-path for users.
+
+[no-bundler-in-docs]: https://github.com/rustwasm/book/issues/150
+
+# Stakeholders
+[stakeholders]: #stakeholders
+
+The stakeholders are primarily people that are doing Rust and Wasm development
+without a JavaScript bundler, which potentially will include all of our new
+users who are just starting the Game of Life tutorial. Most of this demographic
+is probably not watching the RFCs repository, and only some are regularly coming
+to working group meetings, so it makes sense to advertise this RFC in TWiRaWA
+and on our @rustwasm Twitter account.
+
+# Detailed Explanation
+[detailed-explanation]: #detailed-explanation
+
+We will add two new subcommands to `wasm-pack`:
+
+1. `watch`
+2. `serve`
+
+## `wasm-pack watch`
+
+The `watch` subcommand will observe the crate directory for changes and
+automatically re-build when files change.
+
+### Command Line Interface
+
+The `watch` command takes an identical set of command line arguments as
+`wasm-pack build` does.
+
+```
+USAGE:
+    wasm-pack watch [FLAGS] [OPTIONS] [path] [-- <extra_options>...]
+
+FLAGS:
+        --dev              Create a development build. Enable debug info, and disable optimizations.
+        --no-typescript    By default a *.d.ts file is generated for the generated JS file, but this flag will disable
+                           generating this TypeScript file.
+    -h, --help             Prints help information
+        --profiling        Create a profiling build. Enable optimizations and debug info.
+        --release          Create a release build. Enable optimizations and disable debug info.
+    -V, --version          Prints version information
+
+OPTIONS:
+    -m, --mode <mode>          Sets steps to be run. [possible values: no-install, normal, force] [default: normal]
+    -d, --out-dir <out_dir>    Sets the output directory with a relative path. [default: pkg]
+    -s, --scope <scope>        The npm scope to use in package.json, if any.
+    -t, --target <target>      Sets the target environment. [possible values: browser, nodejs, no-modules] [default:
+                               browser]
+
+ARGS:
+    <path>                The path to the Rust crate.
+    <extra_options>...    List of extra options to pass to `cargo build`
+```
+
+### Implementation
+
+We can use [the `notify` crate][notify] to watch the filesystem for
+changes. Initially, we will just watch the crate directory for
```

Comment:
One thing I might recommend is that initially instead of watching just the crate directory we have a heuristic which attempts to find a git repository which contains the crate going up the tree. If found we watch the entire git repository, and if not found we watch the entire crate directory.

All of cargo/wasm-pack/wasm-bindgen are quite speedy when a change was made that doesn't actually change the output artifact, so I think it'd be fine to be a bit more coarse here in terms of changes that we listen for.

Now that being said this is something that can be tweaked at any time, so no need to hold up the RFC on a point like this!